lib/puppet/provider/tacacs_server/cisco.rb

```diff
@@ -133,9 +141,19 @@ def flush
       @tacacs_server = nil
       @property_hash[:ensure] = :absent
     else
-      if @property_hash.empty?
+      # On IOS XR, if the port values change, the entity has to be re-created as the ports
+      # form part of the uniquiness of the item on the device. This is opposed to using
+      # the setters on other platforms for the changing of port values.
+      if @property_hash.empty? ||
+         (Facter.value('operatingsystem').eql?('ios_xr') &&
+          @resource[:port] != @tacacs_server.port.to_i)
+
         # create a new Tacacs Server
-        @tacacs_server = Cisco::TacacsServerHost.new(@resource[:name])
+        create_new
+      end
+
+      if Facter.value('operatingsystem').eql?('ios_xr')
```

Comment:
Facter calls are kind of expensive (painfully slow) on NX devices. Would it make sense to get this during `initialize` and cache it there in a class var?